src/euler_types.jl

```diff
@@ -14,7 +14,7 @@ for axis in [:X, :Y, :Z]
     @eval begin
         struct $RotType{T} <: Rotation{3,T}
             theta::T
-            $RotType{T}(theta) where {T} = new{T}(theta)
+            $RotType{T}(theta::Real) where {T} = new{T}(theta)
```

Comment:
This change seems unrelated and would break e.g. interop with `SymPy.Sym`. Please revert.

Reply:
oops! haha I left that in by accident. There are some problems with mean() in its current form that I was trying to address. Namely that you cannot take the mean of two or three axis rotations. I'll file an issue for that!

Reply:
and I'll definitely add a test. I'm a little slammed right now but in a day or two!